cdap-master/src/main/java/co/cask/cdap/data/runtime/main/ExploreCustomClassLoaderTwillRunnable.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *  
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.runtime.main;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import org.apache.twill.api.AbstractTwillRunnable;
+import org.apache.twill.api.Command;
+import org.apache.twill.api.TwillContext;
+import org.apache.twill.api.TwillRunnable;
+import org.apache.twill.api.TwillRunnableSpecification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.CompoundEnumeration;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+
+/**
+ * Used to load a {@link ExploreServiceTwillRunnable} using {@link CustomResourcesClassLoader}. This gives precedence 
+ * to config files bundled with the application over config files in the classpath used to start the application.
+ * This is required as on some clusters the classpath used to start a container contains a stripped down version of 
+ * Hadoop config files.
+ */
+public class ExploreCustomClassLoaderTwillRunnable extends AbstractTwillRunnable {
+  private static final Logger LOG = LoggerFactory.getLogger(ExploreCustomClassLoaderTwillRunnable.class);
+  private static final String CLASS_NAME = ExploreServiceTwillRunnable.class.getName();
+  
+  private TwillRunnable twillRunnable;
+  private ClassLoader customClassLoader;
+
+  public ExploreCustomClassLoaderTwillRunnable(AbstractTwillRunnable twillRunnable) {
+    this.twillRunnable = twillRunnable;
```

Comment:
Better to just take a `TwillRunnableSpecification` from the constructor to avoid confusion that the given `TwillRunnable` is actually never used in runtime.